VungleAdapter/src/main/java/com/chartboost/mediation/vungleadapter/VungleAdapter.kt

```diff
@@ -463,155 +395,86 @@ class VungleAdapter : PartnerAdapter {
+                                "Message: ${adError.message}"
+                    )
+                }
 
-                                override fun onAdViewed(id: String) {
-                                    PartnerLogController.log(DID_TRACK_IMPRESSION)
-                                    showingBanners.add(request.partnerPlacement)
-                                    listener.onPartnerAdImpression(
-                                        createPartnerAd(null, request)
-                                    )
-                                }
-                            })
+                override fun onAdImpression(baseAd: BaseAd) {
+                    PartnerLogController.log(DID_TRACK_IMPRESSION)
+                    listener.onPartnerAdImpression(
+                        createPartnerAd(baseAd, request)
+                    )
+                }
 
-                        ad?.let {
-                            PartnerLogController.log(LOAD_SUCCEEDED)
-                            resumeOnce(
-                                Result.success(createPartnerAd(it, request))
-                            )
-                        } ?: run {
-                            PartnerLogController.log(LOAD_FAILED, "Placement: $placementId.")
-                            resumeOnce(
-                                Result.failure(
-                                    ChartboostMediationAdException(
-                                        ChartboostMediationError.CM_LOAD_FAILURE_UNKNOWN
-                                    )
-                                )
-                            )
-                        }
-                    }
+                override fun onAdLeftApplication(baseAd: BaseAd) {}
 
-                    override fun onError(placementId: String, exception: VungleException) {
-                        PartnerLogController.log(
-                            LOAD_FAILED,
-                            "Placement: $placementId. Error code: ${exception.exceptionCode}. " +
-                                    "Message: ${exception.message}"
-                        )
-                        resumeOnce(
-                            Result.failure(
-                                ChartboostMediationAdException(
-                                    getChartboostMediationError(
-                                        exception
-                                    )
-                                )
-                            )
-                        )
-                    }
+                override fun onAdLoaded(baseAd: BaseAd) {
+                    PartnerLogController.log(LOAD_SUCCEEDED)
+                    resumeOnce(
+                        Result.success(createPartnerAd(vungleBanner.getBannerView(), request))
+                    )
                 }
-            )
+
+                override fun onAdStart(baseAd: BaseAd) {}
+            }
+
+            vungleBanner.load(adm)
```

Comment:
Yep, it's ok. There's a comment at the beginning of the method

```
        // Vungle needs a nullable adm instead of an empty string for non-programmatic ads.
```